Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -24,12 +24,34 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
     private let BackForwardListCellIdentifier = "BackForwardListViewController"
     private var profile: Profile
     private lazy var sites = [String: Site]()
-    private lazy var tableView = UITableView()
     private var isPrivate: Bool
     private var dismissing = false
+    private var currentRow = 0
+    lazy var tableView = UITableView()
+    
+    lazy var shadow: UIView = {
+        let shadow = UIView()
+        shadow.backgroundColor = UIColor(colorLiteralRed: 0, green: 0, blue: 0, alpha: 0.2)
+        return shadow
+    }()
+    
     var tabManager: TabManager!
+    var bvc: BrowserViewController!
     var listData = [(item:WKBackForwardListItem, type:BackForwardType)]()
     
+    var tableHeight: CGFloat
+    {
+        get {
+            return min(CGFloat(BackForwardViewUX.RowHeight*listData.count), self.view.frame.height/2)
```

Comment:
We've run into problems in the pass with wrapping computed properties over UIKit APIs like this in the past when the caller was not on the main thread. Externally this looks like a basic `CGFloat` when it fact it reaches into UIKit which requires the main thread. I would add an `assert(NSThread.isMainThread(), ...)` check here to catch any places this might happen.